cpu/esp_common/esp-wifi/esp_wifi_netdev.c

```diff
@@ -808,7 +880,10 @@ void esp_wifi_setup (esp_wifi_netdev_t* dev)
     /* TODO */
 #endif
 
-#ifdef MCU_ESP8266
+#ifdef MODULE_ESP_WIFI_AP
+    /* Activate the SoftAP interface */
+    result = esp_wifi_set_mode(WIFI_MODE_AP);
+#elif defined(MCU_ESP8266) && !defined(MODULE_ESP_WIFI_AP)
```

Comment:
```suggestion
#elif defined(MCU_ESP8266)
```
should be sufficient since `MODULE_ESP_WIFI_AP` was already tested here. The `#elif` conditional is only used if `MODULE_ESP_WIFI_AP` is not defined.